vignettes/DSMolgenisArmadillo.Rmd.orig

```diff
@@ -0,0 +1,330 @@
+)
+```
+
+## Usage
+When you start to use Armadillo as a backend for DataSHIELD you can use the `DSMolgenisArmadillo` package for research purposes.
+There is a default workflow in DataSHIELD to do analysis. There are several steps that you need to take.
+
+- [Install](#install)
+- [Authenticate](#authenticate)
+- [Assign data](#assign-data)
+- [Transform data](#transform-data)
+- [Performing analysis](#performing-analysis)
+- [Workspaces](#workspaces)
+
+### Install
+The DataSHIELD Armadillo package is available on the MOLGENIS CRAN (https://registry.molgenis.org/repository/R). You can install it by executing the following code-block:
+
+```{r, install Armadillo DataSHIELD package, eval = FALSE}
+install.packages("DSI")
+install.packages("DSMolgenisArmadillo", repos = "https://registry.molgenis.org/repository/R", dependencies = TRUE)
+```
+
+Make sure you install the DataSHIELD client (`dsBaseClient`) to perform the actual analysis. This needs to be a client which is version 6.0 or higher.
+
+```{r, install the Armadillo DataSHIELD client}
+# install the DataSHIELD client
+install.packages("dsBase", repos = c("http://cran.datashield.org", "http://cran.us.r-project.org"), dependencies = TRUE)
+```
+
+**Experimental versions**
+
+If you are interested in using exeperimental versions you can use our own CRAN repository:
+
+```{r, install an experimental version of the Armadillo DataSHIELD client, eval = FALSE}
+# install experimental Armadillo client
+install.packages("DSMolgenisArmadillo", repos = "https://registry.molgenis.org/repository/r-hosted-snapshots")
+```
+
+Load the necessary packages.
+
+```{r, load the necessary packages}
+library(dsBaseClient)
+library(DSMolgenisArmadillo)
+```
+
+### Authenticate
+The login procedure consists of building a login dataframe and performing the login on the Armadillo server. The important part is to specify the driver. This should be `ArmadilloDriver`. 
+
+We support authentication 2 flows. 
+- as normal user (using oauth on the central authentication server)
+- as superuser (using basic authentication, username and password based)
+
+We are currently only using the oauth flow in production environments.
+- Authenticate using the authentication server
+- Authenticate using the superuser credentials
+
+```{r, authenticate with authentication server}
+# method 1: using oauth on the central server
+# specify server url
+armadillo_url <- "http://localhost:8080"
+
+# get token from central authentication server
+token <- armadillo.get_token(armadillo_url)
+token
+# build the login dataframe
+builder <- DSI::newDSLoginBuilder()
+builder$append(server = "armadillo",
+               url = armadillo_url,
+               user = "",
+               password = "",
+               token = token,
+               table = "gecko/2_1-core-1_0/nonrep",
+               driver = "ArmadilloDriver")
+
+# create loginframe
+logindata <- builder$build()
+logindata
+
+datashield.logout(conns)
```

Comment:
Quitting from lines 72-97 (vignettes/DSMolgenisArmadillo.Rmd.orig) 
Error in datashield.logout(conns) : object 'conns' not found